### PR TITLE
Lineage pull request for: skeleton

### PR DESCRIPTION
**Pull request by cisagovbot**
# Lineage Pull Request #

[Lineage] has created this pull request to incorporate new changes found in an
upstream repository:

Upstream repository: [`https://github.com/cisagov/skeleton-ansible-role.git`](https://github.com/cisagov/skeleton-ansible-role.git)
Remote branch: `HEAD`

Check the changes in this pull request to ensure they won't cause issues with
your project.

## ✅ Pre-approval checklist ##

Remove any of the following that do not apply. If you're unsure about
any of these, don't hesitate to ask. We're here to help!

- [x] All relevant type-of-change labels have been added.
- [ ] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [ ] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

---

**Note:** *You are seeing this because one of this repository's maintainers has
configured [Lineage] to open pull requests.*

For more information:

🛠 [Lineage] configurations for this project are stored in `.github/lineage.yml`

📚 [Read more about Lineage][Lineage]

[//]: # (lineage:metadata:{&quot;slayed&quot;:true})
[Lineage]: https://github.com/cisagov/action-lineage/ "Lineage GitHub Action"

### CODE REVIEW

**Review by dav3r — APPROVED** [review]

**Review by jsf9k — APPROVED** [review]

**Review by dav3r — APPROVED** [review]

**Review by dav3r — APPROVED** [review]

File: meta/main.yml [hunk 1]
```diff
@@ -21,22 +21,23 @@ galaxy_info:
   min_ansible_version: "2.10"
   namespace: cisagov
   platforms:
-    - name: Amazon Linux 2
-      versions:
-        - any
+  # Amazon Linux 2023 does not currently offer yum packages
```
Comment (jsf9k) [inline meta/main.yml]: Amazon Linux 2023 uses `dnf`, not `yum`:
```suggestion
  # Amazon Linux 2023 does not currently offer system packages
```

File: molecule/default/molecule-with-systemd.yml [hunk 2]
```diff
@@ -15,24 +15,17 @@ lint: |
...
-    command: /lib/systemd/systemd
-    image: geerlingguy/docker-debian9-ansible:latest
-    name: debian9-systemd
-    platform: amd64
-    pre_build_image: yes
-    privileged: yes
-    volumes:
-      - /sys/fs/cgroup:/sys/fs/cgroup:rw
+  # Amazon Linux 2023 does not current offer
+  # yum packages which are required for this role
```
Comment (jsf9k) [inline molecule/default/molecule-with-systemd.yml]:
```suggestion
  # Amazon Linux 2023 does not currently offer
  # system packages which are required for this role
```

**Review by jsf9k — CHANGES_REQUESTED** [review]
I suggested a couple of minor changes in comments.  More importantly, please remove the `vars/Amazon.yml` and `tasks/Amazon.yml` files since this Ansible role is not currently supporting Amazon Linux.

**Comment by jmorrowomni** [comment]
> I suggested a couple of minor changes in comments. More importantly, please remove the `vars/Amazon.yml` and `tasks/Amazon.yml` files since this Ansible role is not currently supporting Amazon Linux.

I decided to go the other way and update the Amazon code to use dnf.

File: meta/main.yml [hunk 3]
```diff
@@ -21,22 +21,23 @@ galaxy_info:
   min_ansible_version: "2.10"
   namespace: cisagov
   platforms:
-    - name: Amazon Linux 2
+  # Amazon Linux 2023 does not currently offer yum packages
+  # which are required for this role
```
Comment (jsf9k) [inline meta/main.yml]: Please delete this now-incorrect comment:
```suggestion
```

File: molecule/default/molecule-with-systemd.yml [hunk 4]
```diff
@@ -15,19 +15,12 @@ lint: |
   ansible-lint
   flake8
 platforms:
+  # Amazon Linux 2023 does not current offer
+  # yum packages which are required for this role
```
Comment (jsf9k) [inline molecule/default/molecule-with-systemd.yml]: Please delete this now-incorrect comment:
```suggestion
```

File: molecule/default/tests/test_default.py [hunk 5]
```diff
@@ -16,10 +16,8 @@ def test_packages(host):
     distribution = host.system_info.distribution
     if distribution in ["debian", "kali", "ubuntu"]:
         assert host.package("unattended-upgrades").is_installed
-    elif distribution in ["fedora"]:
+    elif distribution in ["fedora", "amzn"]:
```
Comment (jsf9k) [inline molecule/default/tests/test_default.py]: Please alphabetize the platform names in the array.

File: molecule/default/tests/test_default.py [hunk 6]
```diff
@@ -30,10 +28,8 @@ def test_service_enabled(host):
     distribution = host.system_info.distribution
     if distribution in ["debian", "kali", "ubuntu"]:
         assert host.service("unattended-upgrades").is_enabled
-    elif distribution in ["fedora"]:
+    elif distribution in ["fedora", "amzn"]:
```
Comment (jsf9k) [inline molecule/default/tests/test_default.py]: Please alphabetize the platform names in the array.

File: molecule/default/tests/test_default.py [hunk 7]
```diff
@@ -74,20 +70,13 @@ def test_service_configuration(host):
         # There should be three such lines.
         full_command = f"test \"$(awk '{awk_command}' {filename} | sed '{comment_regex}' | grep --invert-match --ignore-case --fixed-strings security | wc --lines) -eq 3\""
         assert host.run(full_command).succeeded
-    elif distribution in ["fedora"]:
+    elif distribution in ["fedora", "amzn"]:
```
Comment (jsf9k) [inline molecule/default/tests/test_default.py]: Please alphabetize the platform names in the array.

File: vars/Amazon.yml
Comment (jsf9k) [inline vars/Amazon.yml:1]: Please delete this file, since now Amazon Linux 2023 can be treated like any other RedHat platform.

File: tasks/Amazon.yml
Comment (jsf9k) [inline tasks/Amazon.yml:1]: Please delete this file, since now Amazon Linux 2023 can be treated like any other RedHat platform.

**Review by jsf9k — CHANGES_REQUESTED** [review]
I like this approach better, but there are still a few things that need to be addressed.

File: molecule/default/molecule-with-systemd.yml (hunk 2 above)
Comment (jsf9k) [reply molecule/default/molecule-with-systemd.yml]: Resolved in favor of [this comment](https://github.com/cisagov/ansible-role-automated-security-updates/pull/30#discussion_r1189072317).

File: meta/main.yml (hunk 1 above)
Comment (jsf9k) [reply meta/main.yml]: Resolved in favor of [this comment](https://github.com/cisagov/ansible-role-automated-security-updates/pull/30#discussion_r1189071878).

**Review by jsf9k — APPROVED** [review]

**Review by dav3r — APPROVED** [review]

**Pull request merged**